pkg/trace/sampler/catalog.go

```diff
@@ -70,22 +76,27 @@ func (cat *serviceKeyCatalog) register(svcSig ServiceSignature) Signature {
 
 // ratesByService returns a map of service signatures mapping to the rates identified using
 // the signatures.
-func (cat *serviceKeyCatalog) ratesByService(localRates, remoteRates map[Signature]float64, defaultRate float64) map[ServiceSignature]float64 {
-	rbs := make(map[ServiceSignature]float64, len(localRates)+1)
+func (cat *serviceKeyCatalog) ratesByService(localRates map[Signature]float64, remoteRates map[Signature]rm, defaultRate float64) map[ServiceSignature]rm {
+	rbs := make(map[ServiceSignature]rm, len(localRates)+1)
 	cat.mu.Lock()
 	defer cat.mu.Unlock()
 	for key, el := range cat.items {
 		sig := el.Value.(catalogEntry).sig
-		// todo:raphael distinguish remote rates from local rates with a separate priority value
-		if rate, ok := remoteRates[sig]; ok {
-			rbs[key] = rate
+		if r, ok := remoteRates[sig]; ok {
+			rbs[key] = r
 		} else if rate, ok := localRates[sig]; ok {
-			rbs[key] = rate
+			rbs[key] = rm{
+				r: rate,
+				m: 1,
```

Comment:
keep to 0 for local rates (don't set) + don't set in the json
- json size won't be impacted if remote is not enabled
- there's no need to give a value as it's already the default

Reply:
They are not sent actually, you can see [here](https://github.com/DataDog/datadog-agent/pull/10359/files#diff-c346290f514dd944d2568ebb92edb7df9bf88e4f84f5b676da9664ae03d68bf3R99).

I just wanted to keep the actual mechanism (`1` is for agent rates). I can change it to `0`, tho.